integration-tests/test/z-batch-serialization.spec.ts

```diff
@@ -0,0 +1,38 @@
+import { sequencerBatch, BatchType } from '@eth-optimism/core-utils'
+
+import { expect } from './shared/setup'
+import { OptimismEnv } from './shared/env'
+import { envConfig } from './shared/utils'
+
+describe('Batch Serialization', () => {
+  let env: OptimismEnv
+  // Allow for each type to be tested. The env var here must be
+  // the same value that is passed to the batch submitter
+  const batchType = envConfig.BATCH_SUBMITTER_SEQUENCER_BATCH_TYPE.toUpperCase()
+  before(async () => {
+    env = await OptimismEnv.new()
+  })
+
+  it('should fetch batches', async () => {
+    const tip = await env.l1Provider.getBlockNumber()
+    const logs = await env.ctc.queryFilter(
+      env.ctc.filters.TransactionBatchAppended(),
+      0,
+      tip
+    )
+    // collect all of the batches
+    const batches = []
+    for (const log of logs) {
+      const tx = await env.l1Provider.getTransaction(log.transactionHash)
+      batches.push(tx.data)
+    }
+
+    expect(batches.length).to.be.gt(0, 'Submit some batches first')
+
+    // decode all of the batches
+    for (const batch of batches) {
+      const decoded = sequencerBatch.decode(batch)
```

Comment:
This should use the new `SequencerBatch` class instead of the old `sequencerBatch` object